python/nammu/controller/AtfAreaController.py

```diff
@@ -15,16 +15,24 @@ class AtfAreaController(object):
     def __init__(self, mainControler):
         # Create view with a reference to its controller to handle events
         self.view = AtfAreaView(self)
-
         # Will also need delegating to parent presenter
         self.controller = mainControler
-
         # Get a reference to the view's undo_manager
         self.undo_manager = self.view.undo_manager
 
+
     def setAtfAreaText(self, text):
+        '''
+        Need to manually make the edit compound start because the setText()
+        method triggers 2 consecutive INSERT events: one to set the text to "" 
+        and another to set the text to the given text.
+        Default behaviour of the compound edit would create a new edit for each
+        INSERT event, so we have to manually group those two together.
+        '''
+        self.view.edit_listener.compound()
```

Comment:
Would be nice if this was a context manager (with self.view.edit_listener.compound() as bla) but I guess this is's supported by Jython/Swing